docs/design/lexical_conventions/README.md

```diff
@@ -33,4 +33,11 @@ A _lexical element_ is one of the following:
 -   a [symbolic token](symbolic_tokens.md)
 
 The sequence of lexical elements is formed by repeatedly removing the longest
-initial sequence of characters that forms a valid lexical element.
+initial sequence of characters that forms a valid lexical element, with the
+following exception:
+
+-   When a numeric literal immediately follows a `.` or `->` symbol, with no
+    intervening whitespace, a real literal is never formed. Instead, the token
+    will end no later than the next `.` character. For example, `tuple.1.2` is
+    five tokens, `tuple` `.` `1` `.` `2`, not three tokens, `tuple` `.` `1.2`.
+    However, `tuple . 1.2` is lexed as three tokens.
```

Comment:
Hmm, I feel like there is a slight clarification that might be needed here:

```suggestion
-   When a numeric literal immediately follows a `.` (that isn't itself part of a numeric literal) or `->` symbol, with no
    intervening whitespace, a real literal is never formed. Instead, the token
    will end no later than the next `.` character. For example, `tuple.1.2` is
    five tokens, `tuple` `.` `1` `.` `2`, not three tokens, `tuple` `.` `1.2`.
    However, `tuple . 1.2` is lexed as three tokens.
```

Not sure this is quite right, and I don't think it's strictly needed -- we'd never consider such a `.` because it'd be consumed forming the numeric literal that contains it, and so it may not even technically be a symbol... But I feel like it's too easy to imagine an ambiguity here, and so might be worth making it a bit more explicit.

Reply:
Replaced "symbol" with "token", to avoid a possible misunderstanding that the `.` might be part of some other token. Does that work for you?